File: to-do app/todo-java/src/main/java/com/renan/todo/business/MailService.java
package com.renan.todo.business;

import java.util.List;

/**
 * Service for sending e-mail for the users
 */
public interface MailService {

    /**
     * Send email to the user (destination)
     *
     * @param destination - user email
     * @param subject     - email subject
     * @param message     - message to be sent
     * @throws BusinessException - In case of any fail
     */
    public void sendEmail(String destination, String subject, String message) throws BusinessException;

    /**
     * Send email to the users (destinationList)
     *
     * @param destinationList - list of users emails
     * @param subject         - email subject
     * @param message         - message to be sent
     * @throws BusinessException - In case of any fail
     */
    public void sendEmail(List<String> destinationList, String subject, String message) throws BusinessException;

    /**
     * Send email to the user to reset the password
     *
     * @param destination     - user email
     * @param newPasswordCode - the code to change the password
     * @throws BusinessException - In case of any fail
     */
    public void sendNewPasswordEmail(String destination, String newPasswordCode) throws BusinessException;

    /**
     * Send email to the user to activate him on the systema
     *
     * @param destination    - user email
     * @param activationCode - the code to activate the user
     * @throws BusinessException - In case of any fail
     */
    public void sendActivationEmail(String destination, String activationCode) throws BusinessException;
}

File: to-do-app/to-do-kotlin/src/main/kotlin/com/renan/todo/service/MailService.kt
package com.renan.todo.service

/**
 * Service for sending e-mail for the users
 */
interface MailService {

    /**
     * Send email to the user (destination)
     */
    @Throws(BusinessException::class)
    fun sendEmail(destination: String, subject: String, message: String)

    /**
     * Send email to the users (destinationList)
     */
    @Throws(BusinessException::class)
    fun sendEmail(destinationList: List<String>, subject: String, message: String)

    /**
     * Send email to the user to reset the password
     */
    @Throws(BusinessException::class)
    fun sendNewPasswordEmail(destination: String, newPasswordCode: String)

    /**
     * Send email to the user to activate him on the system
     */
    @Throws(BusinessException::class)
    fun sendActivationEmail(destination: String, activationCode: String)

}